src/main/java/core/basesyntax/dao/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,47 @@
+package core.basesyntax.dao;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.lib.Dao;
+import core.basesyntax.model.Manufacturer;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        Storage.addManufacturer(manufacturer);
+        return manufacturer;
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        return Storage.manufacturers.stream()
+                .filter(s -> s.getId().equals(id))
+                .findFirst();
+    }
+
+    @Override
+    public List<Manufacturer> getAll() {
+        return Storage.manufacturers;
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        Manufacturer newManufacturer = get(manufacturer.getId()).get();
+        newManufacturer.setName(manufacturer.getName());
+        newManufacturer.setCountry(manufacturer.getCountry());
+        return newManufacturer;
```

Comment:
we should iterate the list to find the needed object, current approach won't work correctly when you delete element from list and then try to update it